common-kafka/src/main/java/com/cerner/common/kafka/consumer/ProcessingKafkaConsumer.java

```diff
@@ -755,17 +755,19 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitionsAssigned)
             LOGGER.info("Rebalance [{}] is in progress. Old partitions: {} New Partitions: {}", REBALANCE_COUNTER.count(),
                     partitions.keySet(), partitionsAssigned);
 
-            // Close all partitions we will no longer use
+            /**
+             * We must reset the local partition state (cached offsets) for all partitions,
+             * even if the last assignment we saw also included these partitions,
+             * because we may have been disconnected and missed an entire group generation
+             * during which another consumer might have been assigned the partition and committed offsets.
+             */
             partitions.values().stream()
```

Comment:
The `.stream()` is technically not needed now. Maybe create a private `closePartitions()` method and call there here and in `close()`.

Reply:
I missed the second part of this comment earlier. I've now added a private method that both these places call.